oioswift/common/middleware/autocontainerbase.py

```diff
@@ -71,6 +72,11 @@ def _extract_path(self, path):
 
         return account, container, obj
 
+    def _save_bucket(self, env):
+        req = Request(env)
+        account, container, obj = self._extract_path(req.path_info)
+        req.headers['oio-bucket'] = container
```

Comment:
It would be better to save this as an "object meta" or "object sysmeta" instead of using a hack to let this header pass through.